response.go

```diff
@@ -487,6 +427,155 @@ func toShallowNode(node *Node) *Node {
 	}
 }
 
+func marshalAttribute(node *Node, fieldValue reflect.Value, args []string) error {
+	var omitEmpty, iso8601, rfc3339 bool
+
+	var err error
+
+	if len(args) > 2 {
+		for _, arg := range args[2:] {
+			switch arg {
+			case annotationOmitEmpty:
+				omitEmpty = true
+			case annotationISO8601:
+				iso8601 = true
+			case annotationRFC3339:
+				rfc3339 = true
+			}
+		}
+	}
+
+	if node.Attributes == nil {
+		node.Attributes = make(map[string]interface{})
+	}
+
+	if fieldValue.Type() == reflect.TypeOf(time.Time{}) {
+		err = marshalHandleTime(node, fieldValue, args, iso8601, rfc3339, omitEmpty)
+	} else if fieldValue.Type() == reflect.TypeOf(new(time.Time)) {
+		err = marshalHandleNewTime(node, fieldValue, args, iso8601, rfc3339, omitEmpty)
+	} else if fieldValue.Kind() == reflect.Ptr {
+		err = marshalHandlePtr(node, fieldValue, args, omitEmpty)
+	} else if fieldValue.Kind() == reflect.Slice {
+		err = marshalHandleSlice(node, fieldValue, args, omitEmpty)
+	} else {
+		err = marshalHandleDefault(node, fieldValue, args, omitEmpty)
+	}
```

Comment:
_Long, relaxing sigh of relief not having to scroll up and down the old if statements_

Reply:
😄